diff --git a/setup.py b/setup.py
index 2ea0d0b..895c69a 100644
--- a/setup.py
+++ b/setup.py
@@ -20,13 +20,9 @@ setup(
             extra_compile_args={
                 'cxx': ['-O3'],
                 'nvcc': [
-                    '-O3',
+                    '-O3', '-gencode=arch=compute_120,code=sm_120', '-gencode=arch=compute_120,code=compute_120',
                     # '-gencode=arch=compute_60,code=sm_60',
                     # '-gencode=arch=compute_61,code=sm_61',
-                    '-gencode=arch=compute_70,code=sm_70',
-                    '-gencode=arch=compute_75,code=sm_75',
-                    '-gencode=arch=compute_80,code=sm_80',
-                    '-gencode=arch=compute_86,code=sm_86',
                 ],
             },
         ),
@@ -55,13 +51,9 @@ setup(
             extra_compile_args={
                 'cxx': ['-O2'],
                 'nvcc': [
-                    '-O2',
+                    '-O2', '-gencode=arch=compute_120,code=sm_120', '-gencode=arch=compute_120,code=compute_120',
                     # '-gencode=arch=compute_60,code=sm_60',
                     # '-gencode=arch=compute_61,code=sm_61',
-                    '-gencode=arch=compute_70,code=sm_70',
-                    '-gencode=arch=compute_75,code=sm_75',
-                    '-gencode=arch=compute_80,code=sm_80',
-                    '-gencode=arch=compute_86,code=sm_86',
                 ],
             },
         ),
diff --git a/src/altcorr_kernel.cu b/src/altcorr_kernel.cu
index 9e0c5f7..260c1e3 100644
--- a/src/altcorr_kernel.cu
+++ b/src/altcorr_kernel.cu
@@ -304,7 +304,7 @@ std::vector<torch::Tensor> altcorr_cuda_forward(
   const dim3 threads(BLOCK_H, BLOCK_W);
 
 
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(fmap1.type(), "altcorr_forward_kernel", ([&] {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(fmap1.scalar_type(), "altcorr_forward_kernel", ([&] {
     altcorr_forward_kernel<scalar_t><<<blocks, threads>>>(
         fmap1.packed_accessor32<scalar_t,4,torch::RestrictPtrTraits>(),
         fmap2.packed_accessor32<scalar_t,4,torch::RestrictPtrTraits>(),
diff --git a/src/correlation_kernels.cu b/src/correlation_kernels.cu
index 866b0b1..0a3b4fd 100644
--- a/src/correlation_kernels.cu
+++ b/src/correlation_kernels.cu
@@ -141,7 +141,7 @@ std::vector<torch::Tensor> corr_index_cuda_forward(
   torch::Tensor corr = torch::zeros(
     {batch_size, 2*radius+1, 2*radius+1, ht, wd}, opts);
 
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(volume.type(), "sampler_forward_kernel", ([&] {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(volume.scalar_type(), "sampler_forward_kernel", ([&] {
     corr_index_forward_kernel<scalar_t><<<blocks, threads>>>(
       volume.packed_accessor32<scalar_t,5,torch::RestrictPtrTraits>(),
       coords.packed_accessor32<float,4,torch::RestrictPtrTraits>(),
@@ -172,7 +172,7 @@ std::vector<torch::Tensor> corr_index_cuda_backward(
   const dim3 threads(BLOCK, BLOCK);
 
 
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(volume.type(), "sampler_backward_kernel", ([&] {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(volume.scalar_type(), "sampler_backward_kernel", ([&] {
     corr_index_backward_kernel<scalar_t><<<blocks, threads>>>(
       coords.packed_accessor32<float,4,torch::RestrictPtrTraits>(),
       corr_grad.packed_accessor32<scalar_t,5,torch::RestrictPtrTraits>(),
diff --git a/thirdparty/lietorch b/thirdparty/lietorch
--- a/thirdparty/lietorch
+++ b/thirdparty/lietorch
@@ -1 +1 @@
-Subproject commit 0fa9ce8ffca86d985eca9e189a99690d6f3d4df6
+Subproject commit 0fa9ce8ffca86d985eca9e189a99690d6f3d4df6-dirty
